File: backend/app/schemas/pagination.py
from typing import Any, Generic, Optional, TypeVar

from pydantic import BaseModel


T = TypeVar("T")


class PageInfo(BaseModel):
    limit: int
    hasNextPage: bool
    nextCursor: Optional[str] = None


class CursorPage(BaseModel, Generic[T]):
    data: list[T]
    pageInfo: PageInfo
    meta: Optional[dict[str, Any]] = None

File: frontend/src/types/pagination.ts
export type PageInfo = {
  limit: number;
  hasNextPage: boolean;
  nextCursor: string | null;
};

export type CursorPage<T> = {
  data: T[];
  pageInfo: PageInfo;
};

export type ListQueryState = {
  limit: number;
  cursor?: string | null;
  sortKey: string;
  sortDir: "asc" | "desc";
  search?: string;
  status?: string;
};
